storage/src/qmdb/any/db.rs

```diff
@@ -0,0 +1,502 @@
+//! A shared, generic implementation of the "Any" QMDB.
+//! The impl block in this file defines shared functionality across all "Any" QMDB variants.
+
+use super::operation::{update::Update, Operation};
+use crate::{
+    index::Unordered as UnorderedIndex,
+    journal::{
+        authenticated,
+        contiguous::{Contiguous, MutableContiguous, PersistableContiguous},
+    },
+    mmr::{
+        mem::{Clean, Dirty, State},
+        Location, Proof,
+    },
+    qmdb::{
+        any::ValueEncoding,
+        build_snapshot_from_log,
+        operation::{Committable, Operation as OperationTrait},
+        store::LogStore,
+        Error, FloorHelper,
+    },
+    AuthenticatedBitMap,
+};
+use commonware_codec::Codec;
+use commonware_cryptography::{Digest, DigestOf, Hasher};
+use commonware_runtime::{Clock, Metrics, Storage};
+use commonware_utils::Array;
+use core::{num::NonZeroU64, ops::Range};
+use tracing::debug;
+
+/// Type alias for the authenticated journal used by [Db].
+pub(crate) type AuthenticatedLog<E, C, H, S = Clean<DigestOf<H>>> =
+    authenticated::Journal<E, C, H, S>;
+
+/// An "Any" QMDB implementation generic over ordered/ordered keys and variable/fixed values.
+/// Consider using one of the following specialized variants instead, which may be more ergonomic:
+/// - [crate::qmdb::any::ordered::Fixed]
+/// - [crate::qmdb::any::ordered::Variable]
+/// - [crate::qmdb::any::unordered::Fixed]
+/// - [crate::qmdb::any::unordered::Variable]
+pub struct Db<
```

Comment:
I'm not sure I'm fond of calling this structure _Db_ but could be convinced. I suppose since it isn't public it's not that important.